pkg/controller/validate/validate.go

```diff
@@ -155,32 +160,42 @@ func moreThanOneIsNonNil(values ...interface{}) bool {
 }
 
 func instanceSizeForAdvancedDeployment(replicationSpecs []*mdbv1.AdvancedReplicationSpec) error {
-	var instanceSize string
-	err := errors.New("instance size must be the same for all nodes in all regions and across all replication specs for advanced deployment ")
+	err := errors.New("instance size must be the same for all nodes in all regions and across all replication specs for advanced deployment")
+
+	instanceSize := ""
+	firstNonEmptySize := func(region *mdbv1.AdvancedRegionConfig) string {
```

Comment:
nitpick, I see no reason why this cannot be a regular external private function.